### PR TITLE
[ONNX] Gemm operator

### PR DESCRIPTION
**Pull request by tsocha**
merge after: https://github.com/NervanaSystems/ngraph/pull/1472
- Mul operator
- Gemm operator
- Reshape helper

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/gemm.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                auto input_b = inputs.at(1);
+                auto input_c = inputs.at(2);
+
+                double alpha{node.get_attribute_value<double>("alpha", 1)};
+                double beta{node.get_attribute_value<double>("beta", 1)};
+
+                auto trans_a{node.get_attribute_value<int64_t>("transA", 0)};
+                auto trans_b{node.get_attribute_value<int64_t>("transB", 0)};
+
+                if (trans_a)
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/gemm.cpp]: Code style
```
if (<expr>)
{
    ...
}
```

File: src/ngraph/frontend/onnx_import/op/gemm.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    input_a = transpose(input_a);
+                if (trans_b)
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/gemm.cpp]: Same here

File: src/ngraph/frontend/onnx_import/op/gemm.hpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+#include "ngraph/node_vector.hpp"
+#include "ngraph/op/add.hpp"
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/dot.hpp"
+#include "ngraph/op/multiply.hpp"
+
+#include "ngraph/frontend/onnx_import/utils/broadcasting.hpp"
+#include "ngraph/frontend/onnx_import/utils/reshape.hpp"
+
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/gemm.hpp]: This file does not need all of the heraders listed above. The only needed are `ngraph/frontend/onnx_import/node.hpp`, `ngraph/node_vector.hpp`. Move rest of the headers to the .cpp file

File: src/ngraph/frontend/onnx_import/utils/reshape.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#include "ngraph/node_vector.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        std::shared_ptr<ngraph::Node>
+            reorder_axes(std::shared_ptr<ngraph::Node>& node,
+                         std::vector<size_t> axes_order = std::vector<size_t>())
+        {
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/utils/reshape.cpp]: `axes_order = {}` should be enough

File: src/ngraph/frontend/onnx_import/utils/reshape.cpp [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+#include "ngraph/frontend/onnx_import/tensor.hpp"
+#include "ngraph/frontend/onnx_import/utils/reshape.hpp"
+#include "ngraph/node_vector.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        std::shared_ptr<ngraph::Node>
+            reorder_axes(std::shared_ptr<ngraph::Node>& node,
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/utils/reshape.cpp]: std::pass shared_ptr<> as const reference or by value.

File: src/ngraph/frontend/onnx_import/utils/reshape.hpp [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+#pragma once
+
+#include "ngraph/node_vector.hpp"
+#include "ngraph/op/constant.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        std::shared_ptr<ngraph::Node> reorder_axes(const std::shared_ptr<ngraph::Node>& node,
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/utils/reshape.hpp]: Please add some documentation.

File: src/ngraph/frontend/onnx_import/utils/reshape.hpp [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+                                                   std::vector<int> axes_order);
+        std::shared_ptr<ngraph::Node> transpose(const std::shared_ptr<ngraph::Node>& node);
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/utils/reshape.hpp]: Please add some documentation.

File: src/ngraph/frontend/onnx_import/op/gemm.cpp [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+                    input_b = transpose(input_b);
+                }
+
+                //code from python not implemented in c++ yet.
+                //reshape_for_matmul(node, input_a, input_b);
+
+                std::shared_ptr<ngraph::Node> a_dot_b =
+                    std::make_shared<ngraph::op::Dot>(input_a, input_b);
+
+                std::shared_ptr<ngraph::Node> alpha_node = std::make_shared<ngraph::op::Constant>(
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/gemm.cpp]: Let's make a `make_constant` helper utility function.

File: src/ngraph/frontend/onnx_import/op/gemm.cpp [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+                    a_dot_b->get_element_type(), ngraph::Shape{}, std::vector<double>{alpha});
+                alpha_node = std::make_shared<ngraph::op::Broadcast>(
+                    alpha_node,
+                    a_dot_b->get_shape(),
+                    calculate_broadcast_axes(a_dot_b->get_shape(), alpha_node->get_shape()));
+                a_dot_b = std::make_shared<ngraph::op::Multiply>(alpha_node, a_dot_b);
+
+                std::shared_ptr<ngraph::Node> beta_node = std::make_shared<ngraph::op::Constant>(
+                    input_c->get_element_type(), ngraph::Shape{}, std::vector<double>{beta});
+                beta_node = std::make_shared<ngraph::op::Broadcast>(
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/op/gemm.cpp]: Let's make a helper function which simplifies this broadcasting, e.g. `broadcast(node, new_shape)`

**Pull request merged**